plugins/com.redhat.fabric8analytics.lsp.eclipse.ui/src/com/redhat/fabric8analytics/lsp/eclipse/ui/Fabric8AnalyticsStreamConnectionProvider.java

```diff
@@ -87,14 +87,17 @@ public void stop() {
 	protected ProcessBuilder createProcessBuilder() {
 		ProcessBuilder res = super.createProcessBuilder();
 		try {
-			token = Fabric8AnalysisPreferences.getInstance().getToken();
-			serverUrl = Fabric8AnalysisPreferences.getInstance().getProdURL();
-			String [] arrOfStr = serverUrl.split("http", 2);
-			serverUrl = "https" + arrOfStr[1];
-			String temp_server_url = "https://recommender.api.openshift.io/api/v1";
-			res.environment().put(RECOMMENDER_API_TOKEN, token);
-			//			res.environment().put(RECOMMENDER_API_URL, serverUrl);
-			res.environment().put(RECOMMENDER_API_URL, temp_server_url);
+			if( Fabric8AnalysisPreferences.getInstance().isLSPServerEnabled())
```

Comment:
The if should not be necessary since the createProcessBuilder is called from start() method of super class (ProcessStreamConnectionProvider) only if the server is enabled. 

if (!Fabric8AnalysisPreferences.getInstance().isLSPServerEnabled()) {
			throw new IOException("Analyses Disabled");
	}
super.start();  -> Here is the createProcessBuilder called.